app/src/features/submissions/list/SubmissionsListSortMenu.tsx

```diff
@@ -1,19 +1,57 @@
-import { mdiChevronDown } from '@mdi/js';
+import { mdiChevronDown, mdiSortAscending, mdiSortDescending, mdiSortReverseVariant } from '@mdi/js';
 import Icon from '@mdi/react';
+import { MenuItem, useTheme } from '@mui/material';
 import Button from '@mui/material/Button';
 import Menu from '@mui/material/Menu';
-import Stack from '@mui/material/Stack';
 import { FuseResult } from 'fuse.js';
 import { SubmissionRecord } from 'interfaces/useSubmissionsApi.interface';
+import sortBy from 'lodash-es/sortBy';
 import React, { useState } from 'react';
-import SortMenuItem from './SortMenuItem';
+import { objectKeys } from 'utils/Utils';
 
 export type SortSubmission = FuseResult<SubmissionRecord> | SubmissionRecord;
 
+type SortProp = { key: keyof SubmissionRecord; sort: 'asc' | 'desc' };
+
 export interface ISubmissionsListSortMenuProps<TSubmission> {
+  /**
+   * Submissions to sort
+   *
+   * @type {TSubmission[]}
+   * @memberof ISubmissionsListSortMenuProps
+   */
   submissions: TSubmission[];
+
+  /**
+   * Callback fired after submissions are sorted
+   *
+   * @param {TSubmission[]} - data
+   * @returns {void}
+   * @memberof ISubmissionsListSortMenuProps
+   */
   handleSubmissions: (data: TSubmission[]) => void;
+
+  /**
+   * Sort menu items
+   * The object values will be the labels for the menu items
+   *
+   * @type {Partial<Record<keyof SubmissionRecord, string>>}
+   * @example: { name: 'Name', publish_timestamp: 'Publish Date' }
+   * @memberof ISubmissionsListSortMenuProps
+   */
   sortMenuItems: Partial<Record<keyof SubmissionRecord, string>>;
+
+  /**
+   * Manually syncs the default sort with what api responds with
```

Comment:
What does it mean to "sync the default sort with what the API responds with"?

Reply:
Yea this was the fix for having the default selected sort be the same as the api response.

ie: The api responds with the submissions sorted differently based on each endpoint (so couldnt just default to submission_timestamp etc). This just manually sets the default selected sort for the menu. This prevents a dead click if the first menu item selected is the same as the initial sort.

Reply:
A better prop name might be 'defaultSelectedSort'

Reply:
Or maybe `initialSelectedSort`

Reply:
We can address this later as tech debt.